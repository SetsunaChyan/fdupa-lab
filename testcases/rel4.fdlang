function main(){
i = 50;
a = 100;
b = 200;

while(i > 0) {
    i = i - 1;
    a = a - 1;
    b = b - 2;
}

c = a - b;

check_interval(i, 0, 0);
check_interval(a, 50, 50);
check_interval(b, 0, 150);
check_interval(b, 100, 100);
check_interval(c, 0, 50);
check_interval(c, 50, 50);
}