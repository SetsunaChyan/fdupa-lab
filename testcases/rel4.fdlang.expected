Line 14: YES; i in [0, 0]
Line 15: YES; a in [50, 50]
Line 16: YES; b in [100, 100]
Line 17: YES; b in [100, 100]
Line 18: YES; c in [0, 0]
Line 19:  NO; c in [0, 0]
